hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleEntityId.java

```diff
@@ -52,7 +57,8 @@ public MerkleEntityId(long shard, long realm, long num) {
 	}
 
 	public static MerkleEntityId fromAccountId(AccountID grpc) {
-		return new MerkleEntityId(grpc.getShardNum(), grpc.getRealmNum(), grpc.getAccountNum());
+		return SELF_SER_BY_GRPC.computeIfAbsent(grpc, ignore ->
+				new MerkleEntityId(grpc.getShardNum(), grpc.getRealmNum(), grpc.getAccountNum()));
```

Comment:
Perhaps a `new MerkleEntityId(AccountID)` constructor that clones might be better here?

Reply:
I think Java constructors _must_ create a new object; the goal of this factory was to avoid re-creating `MerkleEntityId` objects with the same value many times.